### ISSUE TITLE
test step - allow namespace outside of of kubectl command

### ISSUE THREAD

**Issue by jmccormick2001** [kudobuilder/kuttl#14]
- I have a TestStep that calls kubectl, I'm wanting the TestStep to use a namespace for running that kubectl command...is this possible?  so far, I have to add the --namespace onto the kubectl command string itself to have it run within that namespace.
~~~~
apiVersion: kudo.dev/v1alpha1
kind: TestStep
delete:
- apiVersion: v1
  kind: ConfigMap
  name: my-static-test
  namespace: rq
kubectl:
- create configmap my-static-test --from-file=bundle=/tmp/bundle.zip --namespace=rq
~~~~

**Comment by justinbarrick** [comment]
If you omit the namespace there it will append the namespace flag for you.

Alternatively you could use the `commands` setting which has a `namespaced` option (`kubectl` is legacy as it always sets the namespace flag): https://kudo.dev/docs/testing/reference.html#teststep

**Comment by jmccormick2001** [comment]
thanks, do you happen to have an example of using the commands setting?  I can't seem to get it to work, its creating its own namespace instead of using my predefined namespace.

**Comment by jbarrick-mesosphere** [comment]
Oh, you would want to do:

```
apiVersion: kudo.dev/v1alpha1
kind: TestStep
delete:
- apiVersion: v1
  kind: ConfigMap
  name: my-static-test
  namespace: rq
commands:
- command: kubectl create configmap my-static-test --from-file=bundle=/tmp/bundle.zip --namespace=rq
```

**Comment by kensipe** [comment]
@jmccormick2001 Looks like the main issue here is namespace creation... which is a dup of #6 and fixed in #137 .   There is also support for deletes in a namespace.  I would also recommend for commands to use the `$NAMESPACE` expansion.
Closing for now... please reopen if we are missing something.  thanks!
